src/object/C4Shape.cpp

```diff
@@ -273,250 +331,346 @@ bool C4Shape::LineConnect(int32_t tx, int32_t ty, int32_t cvtx, int32_t ld, int3
+				VtxContactCNAT[vertex] |= CNAT_Center;
 				ContactCount++;
-				// Vertex center contact, now check top,bottom,left,right
-				if (CheckTouchableMaterial(x,y-1,cvtx, collide_halfvehic ? 1 : 0))
-					VtxContactCNAT[cvtx]|=CNAT_Top;
-				if (CheckTouchableMaterial(x,y+1,cvtx, collide_halfvehic ? 1 : 0))
-					VtxContactCNAT[cvtx]|=CNAT_Bottom;
-				if (CheckTouchableMaterial(x-1,y,cvtx, collide_halfvehic ? 1 : 0))
-					VtxContactCNAT[cvtx]|=CNAT_Left;
-				if (CheckTouchableMaterial(x+1,y,cvtx, collide_halfvehic ? 1 : 0))
-					VtxContactCNAT[cvtx]|=CNAT_Right;
+				// Vertex center contact, now check top, bottom, left, right
+				// Not using our style guideline here, is more readable in "table" format
+				if (CheckTouchableMaterial(x, y - 1, vertex, collide_halfvehic ? 1 : 0)) VtxContactCNAT[vertex] |= CNAT_Top;
+				if (CheckTouchableMaterial(x, y + 1, vertex, collide_halfvehic ? 1 : 0)) VtxContactCNAT[vertex] |= CNAT_Bottom;
+				if (CheckTouchableMaterial(x - 1, y, vertex, collide_halfvehic ? 1 : 0)) VtxContactCNAT[vertex] |= CNAT_Left;
+				if (CheckTouchableMaterial(x + 1, y, vertex, collide_halfvehic ? 1 : 0)) VtxContactCNAT[vertex] |= CNAT_Right;
 			}
 			if (border_hack_contacts)
 			{
-				if (x == 0 && CheckTouchableMaterial(x-1, y, cvtx)) *border_hack_contacts |= CNAT_Left;
-				else if (x == ::Landscape.GetWidth() && CheckTouchableMaterial(x+1, y, cvtx)) *border_hack_contacts |= CNAT_Right;
+				if (x == 0 && CheckTouchableMaterial(x - 1, y, vertex))
+				{
+					*border_hack_contacts |= CNAT_Left;
+				}
+				else if (x == ::Landscape.GetWidth() && CheckTouchableMaterial(x + 1, y, vertex))
+				{
+					*border_hack_contacts |= CNAT_Right;
+				}
 			}
 		}
+	}
 
 
 	return !!ContactCount;
 }
 
+
 bool C4Shape::CheckScaleToWalk(int x, int y)
 {
 	for (int32_t i = 0; i < VtxNum; i++)
 	{
 		if (VtxCNAT[i] & CNAT_NoCollision)
+		{
 			continue;
+		}
 		if (VtxCNAT[i] & CNAT_Bottom)
 		{
-			// no ground under the feet?
+			// No ground under the feet?
 			if (CheckTouchableMaterial(x + VtxX[i], y + VtxY[i] + 1, i))
+			{
 				return false;
+			}
 		}
 		else
 		{
-			// can climb with hands?
+			// Can climb with hands?
 			if (CheckTouchableMaterial(x + VtxX[i] - 1, y + VtxY[i], i))
+			{
 				return false;
+			}
 			if (CheckTouchableMaterial(x + VtxX[i] + 1, y + VtxY[i], i))
+			{
 				return false;
+			}
 		}
 	}
 	return true;
 }
 
+
 int32_t C4Shape::GetVertexX(int32_t iVertex)
 {
-	if (!Inside<int32_t>(iVertex,0,VtxNum-1)) return 0;
+	if (!Inside<int32_t>(iVertex, 0, VtxNum - 1))
+	{
+		return 0;
+	}
 	return VtxX[iVertex];
 }
 
+
 int32_t C4Shape::GetVertexY(int32_t iVertex)
 {
-	if (!Inside<int32_t>(iVertex,0,VtxNum-1)) return 0;
+	if (!Inside<int32_t>(iVertex, 0, VtxNum - 1))
+	{
+		return 0;
+	}
 	return VtxY[iVertex];
 }
 
-void C4Shape::CopyFrom(C4Shape rFrom, bool bCpyVertices, bool fCopyVerticesFromSelf)
+void C4Shape::CopyFrom(C4Shape shape_from, bool copy_vertices, bool copy_vertices_from_self)
```

Comment:
Just `from` (or `source`) is fine, we know it's a shape from its type.

Reply:
Done